ndsl/dsl/dace/orchestration.py

```diff
@@ -60,22 +60,11 @@ def _download_results_from_dace(
     config: DaceConfig, dace_result: Optional[List[Any]], args: List[Any]
 ):
     """Move all data from DaCe memory space to GT4Py"""
-    gt4py_results = None
-    if dace_result is not None:
-        if config.is_gpu_backend():
-            gt4py_results = [
-                gt4py.storage.from_array(
-                    r,
-                    backend=config.get_backend(),
-                )
-                for r in dace_result
-            ]
-        else:
-            gt4py_results = [
-                gt4py.storage.from_array(r, backend=config.get_backend())
-                for r in dace_result
-            ]
-    return gt4py_results
+    if dace_result is None:
+        return None
+
+    backend = config.get_backend()
+    return [gt4py.storage.from_array(result, backend=backend) for result in dace_result]
```

Comment:
This turned out to be the same for CPU and GPU. It was just obfuscated by the fact that the GPU branch was intended one level more leading to a less compact version of the same code, making it harder (for humans) to spot the duplication.

Reply:
I am not even sure we need this download at all anymore. I think it ends being a no-op because we own the memory....

Reply:
This PR basically consists of safe code transformations and I would feel a bit unsafe to mix it with such a change. How about I log a small investigation issue?

Reply:
Definitely log it only

Reply:
Logged as https://github.com/GEOS-ESM/NDSL/issues/69